### PR TITLE
Setup Github Actions for testing release binary against linux and darwin

### PR DESCRIPTION
**Pull request by gsquared94**
Runs integration tests at 10 AM UTC every day against the release bits of skaffold in ubuntu and macos. Uses minikube instead of kind.

We can later add badges:
release(linux)/badge.svg) release(darwin)/badge.svg)

### CODE REVIEW

**Comment by briandealwis** [comment]
Cool! But why 10am UTC? That’s mid-day for @dgageot!

**Comment by briandealwis** [comment]
I take it back: 10:00 UTC is brilliant since it's early enough to fix same-day.

File: .github/workflows/integration-linux.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    - name: Install GCloud
+      run: |
+        wget -O gcloud.tar.gz https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-276.0.0-linux-x86_64.tar.gz
+        sudo tar -xvf gcloud.tar.gz -C /
+        sudo CLOUDSDK_PYTHON="python2.7" /google-cloud-sdk/install.sh --usage-reporting=false --bash-completion=false --disable-installation-options
+        echo "::add-path::/google-cloud-sdk/bin"
+
+    - name: Configure GCloud with Docker
+      run:  sudo gcloud auth configure-docker
```
Comment (briandealwis) [inline .github/workflows/integration-linux.yml:46]: Q: Does this do more than configure docker-credential-gcloud?  Does it need credentials?

File: .github/workflows/integration-darwin.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        sudo mv container-structure-test /usr/local/bin/
+
+    - name: Setup other files and permissions
+      run: mkdir -p ${HOME}/.m2/ && cp ./hack/maven/settings.xml ${HOME}/.m2/settings.xml
+
+    - name: Install Minikube and start cluster
+      run: |
+        curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-darwin-amd64
+        sudo install minikube /usr/local/bin/minikube
+        minikube start --profile=minikube --driver=virtualbox
```
Comment (briandealwis) [inline .github/workflows/integration-darwin.yml:86]: Would it make sense to use `--vm-driver none`?  https://github.com/store2be/github-actions/blob/master/minikube/src/index.js#L30

File: .github/workflows/integration-linux.yml (hunk 1 above)
Comment (gsquared94) [reply .github/workflows/integration-linux.yml:46]: > I'm new to Github Actions so excuse naïve questions. Can we create reuseable scripts with pluggable values? Wondering if we can create parallel tests to verify compatibility with our minimum supported versions.

I also just started learning about Github Actions last week and figuring things as I go along. I can try to template out some of these versions, let me see if I can find examples in their documentation.

Comment (gsquared94) [reply .github/workflows/integration-linux.yml:46]: > Q: Does this do more than configure docker-credential-gcloud? Does it need credentials?

I based the steps 1:1 against the way we build the existing integration test [Dockerfile](https://github.com/GoogleContainerTools/skaffold/blob/master/deploy/skaffold/Dockerfile.deps). I think this step might be redundant, I'll try testing without it.

File: .github/workflows/integration-darwin.yml (hunk 2 above)
Comment (gsquared94) [reply .github/workflows/integration-darwin.yml:86]: I don't think macos supports `none`.
https://minikube.sigs.k8s.io/docs/drivers/#macos

File: .github/workflows/integration-linux.yml (hunk 1 above)
Comment (gsquared94) [reply .github/workflows/integration-linux.yml:46]: > Q: Does this do more than configure docker-credential-gcloud? Does it need credentials?

To actually answer the question: It doesn't need any creds and I think it only populates the docker `config.json`:

```
{
  "credHelpers": {
    "us.gcr.io": "gcloud",
    "asia.gcr.io": "gcloud",
    "marketplace.gcr.io": "gcloud",
    "gcr.io": "gcloud",
    "eu.gcr.io": "gcloud",
    "staging-k8s.gcr.io": "gcloud"
  }
}
```
Also I subsequently clear out this file:
`echo '{}' > /home/$(whoami)/.docker/config.json` so I don't think that this step is really needed.

Comment (gsquared94) [reply .github/workflows/integration-linux.yml:46]: Tried without the line `sudo gcloud auth configure-docker` and it resulted in a few failures. There's probably additional configuration that this does. I'll leave it as is 😄

**Comment by gsquared94** [comment]
Refer to [this](https://github.com/gsquared94/skaffold/runs/751341267?check_suite_focus=true) run on macos.
The test `TestBuild/custom` is failing:
```
TestBuild/custom: helper.go:194: skaffold build: exit status 1, Generating tags...
         - skaffold-custom -> gcr.io/k8s-skaffold/skaffold-custom:1aa1110
        Found [minikube] context, using local docker daemon.
        Building [skaffold-custom]...
        2020/06/08 19:59:32 Using base gcr.io/distroless/static:latest for github.com/GoogleContainerTools/skaffold/examples/custom
        2020/06/08 19:59:33 Building github.com/GoogleContainerTools/skaffold/examples/custom
        2020/06/08 19:59:36 Loading ko.local/github.com/googlecontainertools/skaffold/examples/custom:d941ec880876191880b5d66995532d68dfffa0ff1d859ff73ad6b22859a55933
        2020/06/08 19:59:36 Loaded ko.local/github.com/googlecontainertools/skaffold/examples/custom:d941ec880876191880b5d66995532d68dfffa0ff1d859ff73ad6b22859a55933
    TestBuild/custom: panic.go:617: couldn't build "skaffold-custom": the custom script didn't produce an image with tag [gcr.io/k8s-skaffold/skaffold-custom:1aa1110]
```
I couldn't figure out why. It's one of two tests that call `docker deploy` directly in a custom script and are the ones that are failing. I suggest skipping over these 2 test cases for now for Darwin integration. 

We define an env variable in the github actions yml file `INTEGRATION_TEST_OS=macos` and skip these two tests by specifying `MarkIntegrationTest(t, SkipForMacOnGithubActions)`

cc. @nkubala

**Pull request merged**